### PR TITLE
chore: add v0.11.0 release notes

### PR DESCRIPTION
**Pull request by jedevc**

### CODE REVIEW

File: .changes/v0.11.0.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+## v0.11.0 - 2024-04-03
+
+
+### 🔥 Breaking Changes
+- Compatability with v0.11 breaks compatability with old server/client combos - upgrade both!
```
Comment (jedevc) [inline .changes/v0.11.0.md:5]: Note: not sure whether we should explicitly call this out.

**Review by helderco — APPROVED** [review]

File: .changes/v0.11.0.md (hunk 1 above)
Comment (jedevc) [reply .changes/v0.11.0.md]: Decided to not call this out - #7013 removes this incompat.

Comment (jedevc) [reply .changes/v0.11.0.md]: 2 doesn't seem at all feasible to me.

1 is possible, but actually still requires the horrible hack in this PR. Will keep hacking away at it.

Comment (jedevc) [reply .changes/v0.11.0.md]: Determined that 1 is not really possible either given current constraints - technically this is doable with a *whole* lot of fallbacks, so I'll implement these in a follow-up, given that this is an edge case in the first place (and it's been broken before), and we shouldn't block the release of the cloud traces work that it dependent on this engine release.

**Pull request merged**